### DISCUSSION TITLE
Feature request - audit trail

### DISCUSSION THREAD

**Discussion by veenone** [dani-garcia/vaultwarden#3753] [Ideas]
Hello,

I have a request. 
I think Vaultwarden is missing 1 essential feature which is audit trail and activity log.
This will be very useful to keep track any changes made in the items by organization member (i.e: when a password of an account is updated)
this way, all changes in the data will be auditable and traceable. this feature should be available for admin / owners

Thank you

Fienan

**Comment by BlackDex** [comment]
You mean

https://github.com/dani-garcia/vaultwarden/blob/3dbfc484a54c41d1759646444b439da06445060b/.env.template#L91..#L94

**Reply by veenone** [reply]
Yes, this is what I meant :)
however, if I manage different organizations, maybe it would be nice if the report can be generated for global level as well.
and also activities beyond each organization also recorded (i.e: password generation event),

Thank you

Fienan

**Reply by BlackDex** [reply]
Password generation events are not logged, and that is not something we can. Those are done fully client side and are not reported back.

A global overview is out off scope in my opinion. If you really want something like that i suggest to query the database your self and link everything that is needed. Also, we would need to make an interface for that in the Vaultwarden Admin interface, which i think is to much for this project.

**Reply by veenone** [reply]
ok clear,
Thank you for your explanation on this matter

**Comment by misje** [comment]
I am familiar with the organisation events, and I can view them in the web interface. However, I would like to ingest them from a file. Is it technically possible to have these events logged to file, or is there a reason one has to authenticate either through the web interface or the API?

**Reply by BlackDex** [reply]
Technically almost everything is possible of course. Though, it's not much of use to output this also to a file.
It only contains uuid's for the most part.

What i think you want is something like SIEM. Which is not something Vaultwarden supports.

You could export them from the database if you really want, but you need to make the joins/links your self.

**Reply by misje** [reply]
The use case is ingesting the entries into a SIEM, actually. As you say, with just UUIDs, it wouldn't be very useful.
